libcob/cconv.c

```diff
@@ -471,3 +471,84 @@ cob_get_collation_by_name (const char *col_name,
 
 	return col_id;
 }
+
+static int
+cob_convert_hex_digit(char h)
+{
+	if (h >= '0' && h <= '9') return h - '0';
+	else if (h >= 'A' && h <= 'F') return 10 + h - 'A';
+	else if (h >= 'a' && h <= 'f') return 10 + h - 'a';
+	else return -1;
+}
+
+static int
+cob_convert_hex_byte(const char *h)
+{
+	int d1 = cob_convert_hex_digit(h[0]);
+	int d2 = cob_convert_hex_digit(h[1]);
+	if (d1 < 0 || d2 < 0) {
+		return -1;
+	} else {
+		return d1 * 16 + d2;
+	}
+}
+
+int
+cob_load_collation (const char *col_name,
+		    cob_u8_t *ebcdic_to_ascii,
+		    cob_u8_t *ascii_to_ebcdic)
+{
+	char hex[512];
+	int n, i;
+	FILE *f;
+
+	f = fopen(col_name, "rb");
```

Comment:
file opening should be done as with configuration files (which I think is prefixing COB_CONFIG_DIR if not absolute [possibly that needs refactoring to be usable, possibly this can just be called)